tests/internal/parser.c

```diff
@@ -123,6 +128,35 @@ int flb_parser_regex_do(struct flb_parser *parser,
                         void **out_buf, size_t *out_size,
                         struct flb_time *out_time);
 
+static char* mock_timezone(char *tz)
+{
+    char *original_tz = getenv("TZ");
+    if (original_tz) {
+        original_tz = strdup(original_tz);
+        if (!original_tz) {
+            perror("Failed to allocate memory for original_tz");
+            exit(EXIT_FAILURE);
+        }
+    }
+
+    setenv("TZ", tz, 1);
+    tzset();
+
+    return original_tz;
+}
+
+static void undo_mock_timezone(char *original_tz)
+{
+    if (original_tz) {
+        setenv("TZ", original_tz, 1);
+    } else {
+        unsetenv("TZ");
+    }
+    tzset();
+    free(original_tz);
```

Comment:
It's up to you, but imo it's easier to keep track of `free`s if they happen in the same scope as the allocation. I think it would make more sense if `undo_mock_timezone` wasn't in charge of freeing `original_tz`, and instead the test did this after using it to undo the mock timezone.